.circleci/config.yml

```diff
@@ -35,8 +35,82 @@ jobs:
             appcenter login --token ${APP_CENTER_TOKEN}
             export apk="$(find ./app/build/outputs/apk/debug -regex '.*.apk')"
             appcenter distribute release -f ${apk} -g ${APP_CENTER_DISTRIBUTION_LIST} --app ${APP_CENTER_APP_NAME} -r "Android" --silent
-      - store_artifacts: # for display in Artifacts: https://circleci.com/docs/2.0/artifacts/ 
+      - store_artifacts: # for display in Artifacts: https://circleci.com/docs/2.0/artifacts/
           path: app/build/reports
           destination: reports
       - store_test_results: # for display in Test Summary: https://circleci.com/docs/2.0/collect-test-data/
-          path: app/build/test-results
\ No newline at end of file
+          path: app/build/test-results
+
+  test:
+    macos:
+      xcode: "11.4.1"
+    working_directory: ~/code
+
+    steps:
+      - checkout
+      - run:
+          name: set ANDROID_SDK_ROOT
+          command: |
+            echo 'export ANDROID_SDK_ROOT=$HOME/android-tools'  >> $BASH_ENV
+
+      - restore_cache:
+          key: android=tools-v1-{{ checksum "install-android-tools.sh" }}-{{ arch }}
+
+      - run:
+          name: install android tools
+          command: |
+            sh install-android-tools.sh
+            echo 'export PATH=$ANDROID_SDK_ROOT/tools/bin:$PATH'  >> $BASH_ENV
+            echo 'export PATH=$ANDROID_SDK_ROOT/tools:$PATH'  >> $BASH_ENV
+            echo 'export PATH=$ANDROID_SDK_ROOT/platform-tools:$PATH'  >> $BASH_ENV
+            echo 'export PATH=$ANDROID_SDK_ROOT/emulator:$PATH'  >> $BASH_ENV
+            source $BASH_ENV
+            sdkmanager --list
+
+      - save_cache:
+          key: android=tools-v1-{{ checksum "install-android-tools.sh" }}-{{ arch }}
+          paths:
+            - /Users/distiller/android-tools
+
+      - run:
+          name: create AVD
+          command: echo no | avdmanager create avd -d "pixel_xl" -n test -k "system-images;android-29;google_apis;x86_64" --force
```

Comment:
Will this get cached? Was curious what the typical set up is for this on automation, seems like it's a big download to run for every time we run UI tests.